### ISSUE TITLE
Option to disable spam filtering

### ISSUE THREAD

**Issue by andrewkdinh** [anonaddy/anonaddy#10]
Are there plans for an option to disable spam filtering for enabled email aliases? I would like to ensure that **all** emails to some aliases are forwarded to my email.

**Comment by anonaddy** [comment]
Some email may get blocked because the sender has an incorrectly configured mail server. The sender of the email may be failing SPF, DMARC or DNS blacklist checks resulting in the email being rejected. The sender should also have correct reverse DNS setup and use a FQDN as their hostname.

The issue with allowing any and all emails (and potential spam) through from incorrectly configured MTAs is that when forwarding these emails on, the AnonAddy server could be blacklisted for sending spam.

If you are not receiving emails from a certain sender, please contact me via email (GPG key available) with details and I can check to see why the messages are being rejected. I can then potentially whitelist the sender.

**Comment by jarthod** [comment]
I'm wondering about this too because I'm having a similar problem with another provider (at least I believe), some emails I received before are no longer received and I don't know why (though it works fine with a gmail address). I am considering AnonAddy but I don't want to have the same problem obviously so even though I understand the responsibility must be on the sender here, when it's an important stuff like banking email you know it's gonna be almost impossible for them to understand and fix their deprecated/broken setup and it would be good to be able to at least see the error (because they are never gonna investigate for real), and ideally whitelist some senders once you know what the problem is and tried to fix it. To make sure you are gonna receive them. Silently blocking some emails without any information or reason for us (the recipient) is very frustrating and would make me loose trust in the service (as it's currently doing with my provider) because I never know when it's gonna stop working and which emails I already missed, etc..

So the questions here are:
- Is there any way to know about the errors and blocked incoming emails?
- Any way to whitelist (ideally) or disable the spam filter entirely? (maybe only on a self-hosted version to avoid the risk for your IPs?)

Thanks !

**Comment by jarthod** [comment]
@ccorneli thanks for your input! sadly I suppose it answers my question: there is no way to see error or disable the filtering at the moment in AnonAddy?

**Comment by willbrowningme** [comment]
If a mail server that is not correctly configured e.g. invalid FQDN, no reverse DNS attempts to send a message to AnonAddy then it will be rejected, this is to prevent large amounts of spam messages being forwarded and damaging the server's reputation. At the moment there isn't any notification or log of these delivery attempts since they are typically rejected during the EHLO or MAIL FROM steps of the connection.

There is currently a failed deliveries page which displays delivery attempts from AnonAddy to your recipient addresses that have been rejected. You also receive an email notification for each of these.

If you are self-hosting then you are in complete control and can remove as many spam prevention measures as you wish, allowing any and all misconfigured mail servers to send messages to your server. You can also set up your own whitelists etc.

I'm not aware of a way to log all delivery attempts to aliases that are rejected other than by attempting to parse the mail log, since these messages are never accepted they are not piped through to the AnonAddy application and cannot be processed there.

**Comment by jarthod** [comment]
Thanks for the confirmation and explanations @willbrowningme! For the sake of completeness, is it the same for content-based spam filtering or DNS blackist ? (in which case the server is properly configured but might be falsly considered as spam)

**Comment by willbrowningme** [comment]
Yes, the mail servers do run some DNS blacklist checks too. They also run Rspamd which checks things such as SPF, DKIM and DMARC failure, it is very rare that a message is rejected due to its content though.

**Comment by jarthod** [comment]
Ok thanks

**Comment by ririko5834** [comment]
How could I disable spam protection because I get errors like this when I want to reply from anonaddy alias
![image](https://user-images.githubusercontent.com/64729293/194594964-34dacb55-d8e9-4980-917d-3ae343fe70d7.png)
![image](https://user-images.githubusercontent.com/64729293/194594999-ba687268-c3e5-4679-9f47-6dca1a488db3.png)
![image](https://user-images.githubusercontent.com/64729293/194595028-bbd8426d-c9a1-467e-b48b-b9fd35dbc1be.png)

**Comment by Asherathe** [comment]
I don't get a large number of emails I should get through AnonAddy, and I believe this is why. I've checked with my email provider, and they don't see the missing emails being received or rejected on their end. I know I should be getting them, because they're coming from senders like my bank, my insurance company, Facebook...and they just never arrive, with no error message. It got so bad about 9 months ago that I started switching to SimpleLogin, which doesn't seem to have this problem. But, in many cases, I can't change the email address I'm using with a company, because I'd have to prove my identity first by responding on an email...which I never get.
